sycl/test-e2e/AtomicRef/work_item_scope.cpp

```diff
@@ -0,0 +1,63 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <iostream>
+#include <sycl/atomic_ref.hpp>
+#include <sycl/detail/core.hpp>
+#include <sycl/usm.hpp>
+
+int main() {
+
+  sycl::device dev;
+  sycl::queue q(dev);
+  auto ctxt = q.get_context();
+
+  // This test does not validate any output
+  // Only that the work_item scope does not error
+  try {
+
+    // Allocate device memory
+    int *data = sycl::malloc_device<int>(1, q);
```

Comment:
This needs to be `free`'ed properly.

Reply:
Good catch. Fixed.

Reply:
> Fixed.

That's why I dislike this part of the SYCL - no RAII for USM :(

Is that common for other e2e tests to "leak" USM on exceptions?